sql/scan.go

```diff
@@ -167,11 +167,7 @@ func (n *scanNode) renderRow() error {
 		n.row = make([]parser.Datum, len(n.render))
 	}
 	for i, e := range n.render {
-		var err error
-		n.row[i], err = parser.EvalExpr(e, n.vals)
-		if err != nil {
-			return err
-		}
+		n.row[i], _ = parser.EvalExpr(e, n.vals)
```

Comment:
Completely ignoring the error returned by `parser.EvalExpr` seems wrong. The error could have indicated some sort of execution failure like type mismatch. I think what we (you?) need to do is to ensure that `n.vals` has an entry for every column. Either the entry is the actual column value or `parser.DNull` if the column was not present and is nullable. If the column was not present and is not nullable it should be left out of `n.vals`.